packages/backend-data/src/factory.ts

```diff
@@ -97,9 +98,9 @@ export class DataFactory implements ConstructFactory<AmplifyData> {
         this.props,
         buildConstructFactoryProvidedAuthConfig(
           props.constructContainer
-            .getConstructFactory<ResourceProvider<AuthResources>>(
-              'AuthResources'
-            )
+            .getConstructFactory<
+              ResourceProvider<AuthResources | ReferenceAuthResources>
+            >('AuthResources')
```

Comment:
by looking at this we should make `defineAuth` and `referenceAuth` mutually exclusive in single app).

Reply:
yes ill update this shortly

Reply:
done in factory several commits ago, resolving